packages/matchbox/src/components/ListBox/useOptionConstructor.js

```diff
@@ -21,13 +26,33 @@ function useOptionConstructor({ options, value, onSelect, open, placeholder }) {
     })(e);
 
     onKey('arrowUp', () => {
-      e.preventDefault();
       if (focused === 0) {
         setFocused(options.length - 1);
       } else {
         setFocused(focused - 1);
       }
     })(e);
+
+    onKey('enter', () => {
+      onSelect(options[focused].props.value);
+    })(e);
+
+    setKeysSoFar(keysSoFar + e.key.toLowerCase());
+    clearKeysSoFarAfterDelay();
+  }
+
+  function clearKeysSoFarAfterDelay() {
+    if (keyClear) {
+      clearTimeout(timeout);
+      setKeyClear(null);
+    }
+
+    setKeyClear(true);
+
+    let timeout = setTimeout(function() {
+      setKeysSoFar('');
+      setKeyClear(false);
+    }, 700);
```

Comment:
Tried to find other examples of listboxes and what timeout they use – this is spot on